src/components/ScreenStack.tsx

```diff
@@ -40,24 +59,47 @@ function ScreenStack(props: ScreenStackProps) {
   });
 
   React.useEffect(() => {
-    if (gestureDetectorBridge) {
-      gestureDetectorBridge.current.stackUseEffectCallback(ref);
-    }
+    gestureDetectorBridge.current.stackUseEffectCallback(ref);
   });
+
+  const isGestureDetectorProviderNotDetected =
+    ScreenGestureDetector.name !== 'GHWrapper' && goBackGesture !== undefined;
+  const isGestureDetectorNotConfiguredProperly =
+    goBackGesture !== undefined &&
+    screensRefs === undefined &&
+    currentScreenId === undefined;
+
+  warnOnce(
+    isGestureDetectorProviderNotDetected,
+    'Cannot detect GestureDetectorProvider in a screen that uses `goBackGesture`. Make sure your navigator is wrapped in GestureDetectorProvider.',
+  );
+  warnOnce(
+    isGestureDetectorNotConfiguredProperly,
+    'Custom Screen Transition require screensRefs and currentScreenId to be provided.',
+  );
```

Comment:
Extract this to some helper function please

Reply:
Changed in: https://github.com/software-mansion/react-native-screens/pull/2449/commits/a875f370ddbd3a71b088ce7cedc87f6e1a22772a